aws/resource_aws_ssm_parameter.go

```diff
@@ -47,35 +51,47 @@ func resourceAwsSsmParameter() *schema.Resource {
 			"key_id": {
 				Type:     schema.TypeString,
 				Optional: true,
-				ForceNew: true,
+				Computed: true,
 			},
 			"overwrite": {
 				Type:     schema.TypeBool,
 				Optional: true,
-				Default:  false,
 			},
+			"allowed_pattern": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": tagsSchema(),
 		},
 	}
 }
 
-func resourceAwsSsmParameterRead(d *schema.ResourceData, meta interface{}) error {
+func resourceAwsSmmParameterExists(d *schema.ResourceData, meta interface{}) (bool, error) {
 	ssmconn := meta.(*AWSClient).ssmconn
 
-	log.Printf("[DEBUG] Reading SSM Parameter: %s", d.Id())
-
-	paramInput := &ssm.GetParametersInput{
-		Names: []*string{
-			aws.String(d.Id()),
-		},
+	resp, err := ssmconn.GetParameters(&ssm.GetParametersInput{
+		Names:          []*string{aws.String(d.Id())},
 		WithDecryption: aws.Bool(true),
+	})
+
+	if err != nil {
+		return false, err
 	}
+	return len(resp.InvalidParameters) == 0, nil
+}
 
-	resp, err := ssmconn.GetParameters(paramInput)
+func resourceAwsSsmParameterRead(d *schema.ResourceData, meta interface{}) error {
+	ssmconn := meta.(*AWSClient).ssmconn
 
+	log.Printf("[DEBUG] Reading SSM Parameter: %s", d.Id())
+
+	resp, err := ssmconn.GetParameters(&ssm.GetParametersInput{
+		Names:          []*string{aws.String(d.Id())},
+		WithDecryption: aws.Bool(true),
+	})
 	if err != nil {
-		return errwrap.Wrapf("[ERROR] Error describing SSM parameter: {{err}}", err)
+		return errwrap.Wrapf("[ERROR] Error getting SSM parameter: {{err}}", err)
```

Comment:
Minor nitpick: It's currently in a lot of the codebase 😢 , but we do not need `errwrap` for returning simple error strings. `fmt.Errorf("error getting SSM parameter: %s", err)` is fine here